SPDY/SPDYSocket.m

```diff
@@ -1152,7 +1167,7 @@ - (NSError *)streamError
         if (error) return CFBridgingRelease(error);
     }
 
-    return nil;
+    return SPDY_SOCKET_ERROR(SPDYSocketTransportError, @"General socket stream error.");
```

Comment:
yup. as there should be. all callers of streamError are expecting an actual error to get returned, and in some cases even end up asserting so (_handleError to _closeWithError).